src/util/AbstractHelper.sol

```diff
@@ -0,0 +1,337 @@
+        if(dolaBal < repayAmount){
+            require(repayAmount <= dolaAmount + dolaBal, "Repay amount exceeds dola amount");
+            DOLA.transferFrom(msg.sender, address(this), repayAmount - DOLA.balanceOf(address(this)));
+        } else {
+            DOLA.transfer(msg.sender, dolaBal - repayAmount);
+        }
+
+        //Repay repayAmount
+        DOLA.approve(address(market), repayAmount);
+        market.repay(msg.sender, repayAmount);
+    }
+
+    /**
+    @notice Sells DBR on behalf of the caller and uses the proceeds along with DOLA from the caller to repay debt, and then withdraws collateral
+    @dev The caller is unlikely to spend all of the DOLA they make available for the function call
+    @param market Market the user wishes to repay debt in
+    @param dolaAmount Maximum amount of dola debt the user is willing to repay
+    @param minDolaFromDbr Minimum amount of DOLA the caller expects to get in return for selling their DBR
+    @param dbrAmountToSell Amount of DBR the caller wishes to sell
+    @param collateralAmount Amount of collateral to withdraw
+    @param deadline Deadline of the signature
+    @param v V parameter of the signature
+    @param r R parameter of the signature
+    @param s S parameter of the signature
+    */
+    function sellDbrRepayAndWithdrawOnBehalf(
+        IMarket market, 
+        uint dolaAmount, 
+        uint minDolaFromDbr,
+        uint dbrAmountToSell, 
+        uint collateralAmount, 
+        uint deadline, 
+        uint8 v, 
+        bytes32 r, 
+        bytes32 s) 
+        external 
+    {
+        //Repay
+        sellDbrAndRepayOnBehalf(market, dolaAmount, minDolaFromDbr, dbrAmountToSell);
+
+        //Withdraw
+        market.withdrawOnBehalf(msg.sender, collateralAmount, deadline, v, r, s);
+
+        //Transfer collateral to msg.sender
+        IERC20(market.collateral()).transfer(msg.sender, collateralAmount);
+    }
+
+    /**
+    @notice Sells DBR on behalf of the caller and uses the proceeds along with DOLA from the caller to repay debt, and then withdraws collateral
+    @dev The caller is unlikely to spend all of the DOLA they make available for the function call
+    @param market Market the user wishes to repay debt in
+    @param dolaAmount Maximum amount of dola debt the user is willing to repay
+    @param minDolaFromDbr Minimum amount of DOLA the caller expects to get in return for selling their DBR
+    @param dbrAmountToSell Amount of DBR the caller wishes to sell
+    @param collateralAmount Amount of collateral to withdraw
+    @param deadline Deadline of the signature
+    @param v V parameter of the signature
+    @param r R parameter of the signature
+    @param s S parameter of the signature
+    */
+    function sellDbrRepayAndWithdrawNativeEthOnBehalf(
+        IMarket market, 
+        uint dolaAmount, 
+        uint minDolaFromDbr,
```

Comment:
`minDolaFromDbr` is sensitive. UI should provide a reasonable value to prevent sandwich attacks. Comments should make note of it for users who interact directly with the contracts.

Reply:
Fixed in [74e7e73](https://github.com/InverseFinance/FiRM/pull/33/commits/74e7e733169f454ecae4b9c771f033b0cdad8d84)